application-analytics-ui/src/main/resources/Analytics/Code/Macros/MostViewedPages.xml

```diff
@@ -0,0 +1,799 @@
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>long|short|default|forbid</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </cache>
+      <code>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>code</name>
+        <number>2</number>
+        <prettyName>Code</prettyName>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>define('mostViewedPagesTranslation', {
+  prefix: 'Analytics.Code.Macros.',
+  keys: [
+    "label",
+    "nb_visits",
+    "nb_hits",
+    "sum_time_spent",
+    "entry_nb_visits",
+    "entry_nb_actions",
+    "bounce_rate",
+    "exit_rate"
+  ]
+});
+
+
+require(['jquery', 'xwiki-l10n!mostViewedPagesTranslation'], function ($, l10n) {
+  function getAttributes(element) {
+    return Array.from(element.attributes).reduce((acc, attribute) =&gt; {
+      acc[attribute.name] = attribute.value;
+      return acc;
+    }, {});
+  }
+  function constructUrl(attributes) {
+    const paramString = $.param({
+      module: 'API',
+      method: 'Actions.getPageTitles',
+      idSite: attributes.id,
+      period: 'range',
+      date: `${attributes.startdate.replaceAll('/', '-')},${attributes.enddate.replaceAll('/', '-')}`, //the data processing could be moved to velocity
```

Comment:
This line is too long
If you need to do this replace because Matomo expects a different format for the date, a proper formatting method should be used
For example https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-ui/src/main/resources/XWiki/Notifications/Code/LegacyNotificationAdministration.xml#L46